tower-util/src/ext/apply.rs

```diff
@@ -0,0 +1,50 @@
+use std::marker::PhantomData;
+
+use futures::{Future, IntoFuture, Poll};
+use tower_service::Service;
+
+/// `Apply` service combinator
+pub struct Apply<T, F, R, Req> {
+    service: T,
+    f: F,
+    r: PhantomData<Fn(Req) -> R>,
```

Comment:
Nit: since this field isn't used, it should probably have a leading `_`.

Reply:
i am not sure, we need `F` which is `Fn(Req, T) -> IntoFuture`. do i miss anything?